### ISSUE TITLE
FAILOVER GROUP handling

### ISSUE THREAD

**Issue by zanebclark** [littleK0i/SnowDDL#50]
**Describe the bug**
Both the `plan` and `apply` functions fail to process an account with a defined FAILOVER GROUP. Additionally, FAILOVER GROUP is not a valid excluded object type. 

**Expected behavior**
I would expect SnowDDL to ignore FAILOVER GROUPS or allow them to be excluded via exclude-object-types

**Attach log**
```
2023-07-25 15:10:43.921 - INFO - Snowflake version = 7.24.2 (BUSINESS_CRITICAL), SnowDDL version = 0.18.0
2023-07-25 15:10:43.921 - INFO - Account = ******, Region = AWS_US_WEST_2
2023-07-25 15:10:43.921 - INFO - Session = 13050683331014878, User = *****
2023-07-25 15:10:43.921 - INFO - Role = ACCOUNTADMIN, Warehouse = None
2023-07-25 15:10:43.921 - INFO - Roles in session = ACCOUNTADMIN,SYSADMIN,SECURITYADMIN
2023-07-25 15:10:43.921 - INFO - ---
Traceback (most recent call last):
  File "C:\Users\zaneb\AppData\Local\Programs\Python\Python38\lib\runpy.py", line 194, in _run_module_as_main   
    return _run_code(code, main_globals, None,
  File "C:\Users\zaneb\AppData\Local\Programs\Python\Python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\zaneb\AppData\Local\Programs\Python\Python38\Scripts\snowddl.exe\__main__.py", line 7, in <module>
  File "C:\Users\zaneb\AppData\Local\Programs\Python\Python38\lib\site-packages\snowddl\app\base.py", line 365, in entry_point
    app.execute()
  File "C:\Users\zaneb\AppData\Local\Programs\Python\Python38\lib\site-packages\snowddl\app\base.py", line 279, in execute
    resolver.resolve()
  File "C:\Users\zaneb\AppData\Local\Programs\Python\Python38\lib\site-packages\snowddl\resolver\abc_resolver.py", line 49, in resolve
    self.existing_objects = self.get_existing_objects()
  File "C:\Users\zaneb\AppData\Local\Programs\Python\Python38\lib\site-packages\snowddl\resolver\abc_role_resolver.py", line 32, in get_existing_objects
    for role_name, grants, future_grants in self.engine.executor.map(self.get_existing_role_grants, existing_roles):
  File "C:\Users\zaneb\AppData\Local\Programs\Python\Python38\lib\concurrent\futures\_base.py", line 619, in result_iterator
    yield fs.pop().result()
  File "C:\Users\zaneb\AppData\Local\Programs\Python\Python38\lib\concurrent\futures\_base.py", line 444, in result
    return self.__get_result()
  File "C:\Users\zaneb\AppData\Local\Programs\Python\Python38\lib\concurrent\futures\_base.py", line 389, in __get_result
    raise self._exception
  File "C:\Users\zaneb\AppData\Local\Programs\Python\Python38\lib\concurrent\futures\thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\Users\zaneb\AppData\Local\Programs\Python\Python38\lib\site-packages\snowddl\resolver\abc_role_resolver.py", line 48, in get_existing_role_grants
    if ObjectType[r['granted_on']] == ObjectType.MATERIALIZED_VIEW and str(r['name']).endswith('IDX_MV_"'):     
  File "C:\Users\zaneb\AppData\Local\Programs\Python\Python38\lib\enum.py", line 387, in __getitem__
    return cls._member_map_[name]
KeyError: 'ACCOUNT'
```

**Comment by littleK0i** [comment]
@zanebclark , I've pushed a basic fix in `0.18.1`. Grants for unsupported object types should be automatically skipped from now on.

However, it might not fix the fundamental problem. SnowDDL should only check grants for roles it created. I am not sure about the source of this extra "GRANT ... ON ACCOUNT" you've got.

The possible reason is active role `ACCOUNTADMIN`. It is highly recommended to create a separate USER and ROLE for SnowDDL: https://docs.snowddl.com/guides/other/admin

Otherwise SnowDDL will try to manage all objects owned by `ACCOUNTADMIN`, including roles created manually.

**Comment by zanebclark** [comment]
@littleK0i , thanks for the rapid reply! Your instincts were accurate. Without upgrading to the new version, using the suggested SnowDDL user and role cleared up the problem.
